src/pages/Fox/components/FoxHeader.tsx

```diff
@@ -0,0 +1,86 @@
+import { Container, Flex, Heading, Stack } from '@chakra-ui/react'
+import { fromAccountId } from '@shapeshiftoss/caip'
+import { useCallback, useEffect, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router'
+import { AccountDropdown } from 'components/AccountDropdown/AccountDropdown'
+import { Display } from 'components/Display'
+import { InlineCopyButton } from 'components/InlineCopyButton'
+import { PageBackButton, PageHeader } from 'components/Layout/Header/PageHeader'
+import { Text } from 'components/Text'
+import { selectPortfolioAccountIdsByAssetIdFilter } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { useFoxPageContext } from '../hooks/useFoxPageContext'
+
+const containerPadding = { base: 6, '2xl': 8 }
+const containerPaddingTop = { base: 0, md: 8 }
+const accountDropdownButtonProps = { variant: 'solid', width: 'full' }
+
+export const FoxHeader = () => {
+  const translate = useTranslate()
+  const { assetId, setAssetAccountId, selectedAssetAccountId } = useFoxPageContext()
+  const accountIdsFilter = useMemo(() => ({ assetId }), [assetId])
+  const accountIds = useAppSelector(state =>
+    selectPortfolioAccountIdsByAssetIdFilter(state, accountIdsFilter),
+  )
+
+  const history = useHistory()
+
+  const handleBack = useCallback(() => {
+    history.push('/explore')
+  }, [history])
+
+  useEffect(() => {
+    if (!accountIds.length) setAssetAccountId(undefined)
+    if (accountIds.length === 1) {
+      setAssetAccountId(accountIds[0])
+    }
+  }, [accountIds, setAssetAccountId])
```

Comment:
q: do we need to have an effect and set this in the header? if the context knows the account number, it can properly select the right accountId (which will always be a single one, since this is only for FOX.ETH)

Reply:
I've been able to remove this effect and manage everything properly in the context (using the 0 account by default)